### PR TITLE
Clarify GPG public key extraction process and update keyserver URIs

### PR DESCRIPTION
**Pull request by lumarel**
I already had twice the issue to make the connection between the `gpg --verify Release.gpg Release` commands output and the `gpg --recv-keys`, so this change extends that part.

And recently we noticed, that the receiving from `hkp://keyserver.ubuntu.com` does not work anymore -> switches this to `hkps://`

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.10/Katello 4.12
* [x] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8)
* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [x] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

### CODE REVIEW

**Review by maximiliankolb — COMMENTED** [review]
Thanks @lumarel I've checked the examples for Debian 11. I'd not add the stdout, but the rest LGTM

File: guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc [hunk 1]
```diff
@@ -17,26 +17,43 @@ $ wget https://deb.debian.org/debian/dists/bullseye/Release.gpg
...
 $ gpg --verify Release.gpg Release
+gpg: Signature made Sa 10 Feb 2024 13:41:18 CET
+gpg:                using RSA key *A7236886F3CCCAAD148A27F80E98404D386FA1D9*
+gpg: Can't check signature: No public key
+gpg: Signature made Sa 10 Feb 2024 13:41:19 CET
+gpg:                using RSA key *4CB50190207B4758A3F73A796ED0E7B82643E131*
+gpg: Can't check signature: No public key
+gpg: Signature made Sa 10 Feb 2024 13:46:58 CET
+gpg:                using RSA key *A4285295FC7B1A81600062A9605C66F00D6C9793*
+gpg: Can't check signature: No public key
```
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc:28]: Yeah, maybe similar to the alternative suggestion in https://github.com/theforeman/foreman-documentation/pull/3024#discussion_r1604775148, you can add "Note the GPG key fingerprint for any missing public GPG keys." or similar.

**Comment by lumarel** [comment]
@maximiliankolb Thank you for the help, I think I figured it out 👍🏻

File: guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc [hunk 2]
```diff
@@ -36,9 +39,11 @@ $ gpg --verify Release.gpg Release
 +
 [options="nowrap" subs="+quotes"]
 ----
-$ gpg --armor --export 0146DC6D4A0B2914BDED34DB648ACFD622F3D138 A7236886F3CCCAAD148A27F80E98404D386FA1D9 A4285295FC7B1A81600062A9605C66F00D6C9793 > debian_11.txt
+$ gpg --armor --export A7236886F3CCCAAD148A27F80E98404D386FA1D9 4CB50190207B4758A3F73A796ED0E7B82643E131 A4285295FC7B1A81600062A9605C66F00D6C9793 > debian_11.txt
 ----
 +
+Ensure that `gpg` returns a `Good signature` message for each signature.
++
 Use this `.txt` file and upload it to {Project} as described below.
```
Comment (AkshayGadhaveRH) [inline guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc:47]: Maybe rephrase this to:

Upload the `.txt` file to {Project}.
For more information, see xref:Importing_a_Custom_GPG_Key_{context}[import the custom GPG public key] into {Project}.

File: guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc [hunk 3]
```diff
@@ -36,9 +39,11 @@ $ gpg --verify Release.gpg Release
...
 
 Continue to xref:Importing_a_Custom_GPG_Key_{context}[import the custom GPG public key] into {Project}.
```
Comment (AkshayGadhaveRH) [inline guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc:49]: Maybe rephrase this to:

```suggestion
Upload the .txt file to {Project}.
For more information, see xref:Importing_a_Custom_GPG_Key_{context}[import the custom GPG public key] into {Project}.
```

Comment (lumarel) [reply guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc]: Sure 👍🏻

File: guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc [hunk 4]
```diff
@@ -36,9 +39,10 @@ $ gpg --verify Release.gpg Release
...
+$ gpg --armor --export A7236886F3CCCAAD148A27F80E98404D386FA1D9 4CB50190207B4758A3F73A796ED0E7B82643E131 A4285295FC7B1A81600062A9605C66F00D6C9793 > debian_11.txt
 ----
 +
-Use this `.txt` file and upload it to {Project} as described below.
-
-Continue to xref:Importing_a_Custom_GPG_Key_{context}[import the custom GPG public key] into {Project}.
+Ensure that `gpg` returns a `Good signature` message for each signature.
++
+Upload the .txt file to {Project}.
+For more information, see xref:Importing_a_Custom_GPG_Key_{context}[import the custom GPG public key] into {Project}.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_extracting-gpg-public-key-fingerprints-from-release-files.adoc:48]:
```suggestion
For more information, see xref:Importing_a_Custom_GPG_Key_{context}[] into {Project}.
```

Asciidoctor will find the title of the anchor automatically.

**Review by maximiliankolb — APPROVED** [review]
Thanks Lukas, LGTM

**Comment by maximiliankolb** [comment]
Merged to "master" and cherry-picked:
   dba651a07b..bc9fcae6eb  3.10 -> 3.10
   9815a9899b..d3321a3022  3.9 -> 3.9
